src/rpc/rawtransaction.cpp

```diff
@@ -878,6 +878,7 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
                         {
                             {RPCResult::Type::STR_HEX, "txid", "The transaction hash in hex"},
                             {RPCResult::Type::BOOL, "allowed", "If the mempool allows this tx to be inserted"},
+                            {RPCResult::Type::NUM, "fee", "Fee provided in this transaction (only present when 'allowed' is true)"},
```

Comment:
* This will put the fee in raw satoshis? I don't think we do this anywhere else.
* Have you seen https://github.com/bitcoin/bitcoin/pull/19093#issuecomment-663963257? Specifically, it would be better for the caller to have the feerate instead of the absolute fee. Otherwise, the caller is forced to implement a witness-aware tx-deserializer to determine the virtual size. `getmempoolentry` returns vsize, weight, as well as a `fees` object.

Reply:
Hmm. Now that I am thinking it makes sense to return `feerate`. Do you want the `fees` to be changed into `feerate`? 

> I am starting to think that it could even return the same info that `getmempoolinfo` returns

Little confused on this one as `getmepoolinfo` returns the following
```
Result:
{
  "loaded": true|false         (boolean) True if the mempool is fully loaded
  "size": xxxxx,               (numeric) Current tx count
  "bytes": xxxxx,              (numeric) Sum of all virtual transaction sizes as defined in BIP 141. Differs from actual serialized size because witness data is discounted
  "usage": xxxxx,              (numeric) Total memory usage for the mempool
  "maxmempool": xxxxx,         (numeric) Maximum memory usage for the mempool
  "mempoolminfee": xxxxx       (numeric) Minimum fee rate in BTC/kB for tx to be accepted. Is the maximum of minrelaytxfee and minimum mempool fee
  "minrelaytxfee": xxxxx       (numeric) Current minimum relay fee for transactions
}
```

Which of these do you think should be included here?

Reply:
Sorry, typo. Should have been getmempool**entry** in the other comment as well

Reply:
This seems like scope creep. I think returning the fee is fine to start with, and unblocks #19339. Could we defer adding more details to a later PR?

Reply:
* Changing the type to `STR_AMOUNT` like in all other places should be done for consistency and to protect against mistakes. I consider this a bug that should be fixed before merge.
* Removing the field `fee` and replacing it with something else later is a breaking rpc change. For the sanity of our users we should keep breaking rpc changes at a minimum.

Reply:
Does #19339 need the rpc changes? If not, then this can trivially be unblocked by cherry-picking this pull and undoing the RPC changes. I think an internal (non-rpc) change can go in earlier, as it doesn't have the breaking rpc change concerns.

Reply:
Good point about cherry picking the non-rpc changes.

Separately, I think `fee` is a useful value to return to the user, since it's exact (unlike feerate which might be rounded).

Reply:
Agree, since `getmempoolentry` also returns the absolute fee (and the vsize)

Reply:
Yeah good point, I only need the fee from ATMP for #19339 so I'll just use that. @rajarshimaitra is listed as coauthor on there, and can continue iterating on the RPC stuff here.